tooling/camel-karaf-feature-maven-plugin/src/test/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojoTest.java

```diff
@@ -0,0 +1,142 @@
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+
+        location = "wrap:mvn:org.apache.olingo/odata-server-core/5.0.0$overwrite=merge";
+        expected = "wrap:mvn:org.apache.olingo/odata-server-core/5.0.0$overwrite=merge&Bundle-Version=5.0.0";
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+
+        location = "wrap:mvn:org.apache.olingo/odata-server-core/5.0.0$overwrite=merge&Export-Package=org.apache.olingo.*;version=5.0.0";
+        expected = "wrap:mvn:org.apache.olingo/odata-server-core/5.0.0$overwrite=merge&Bundle-Version=5.0.0&Export-Package=org.apache.olingo.*;version=5.0.0";
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+
+        // original version won't work in the karaf container
+        location = "wrap:mvn:com.google.apis/google-api-services-storage/v1-rev20240209-2.0.0";
+        expected = "wrap:mvn:com.google.apis/google-api-services-storage/v1-rev20240209-2.0.0$Bundle-Version=0.0.0.v1-rev20240209-2_0_0";
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+
+        // bundle version header is present but it won't work in the karaf container
+        location = "mvn:commons-io/commons-io/2.15.1$Bundle-Version=2.15.0";
+        expected = "mvn:commons-io/commons-io/2.15.1$Bundle-Version=2.15.0";
+        ;
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+
+        // bundle version header is present but it points to the wrong value
+        location = "mvn:commons-io/commons-io/2.15.1$Bundle-Version=2.15.0";
+        expected = "mvn:commons-io/commons-io/2.15.1$Bundle-Version=2.15.0";
+        ;
+        assertEquals(expected, ensureVersionMojo.processLocation(location));
+    }
+
+    @Test
+    void getVersionStartIndexTest() {
+        assertEquals(51,
+                ensureVersionMojo.getVersionStartIndex("wrap:mvn:org.apache.httpcomponents.core5/httpcore5/5.2.1"));
+        assertEquals(51, ensureVersionMojo.getVersionStartIndex(
+                "wrap:mvn:org.eclipse.californium/element-connector/3.11.0$overwrite=merge&Import-Package=net.i2p.crypto.eddsa;resolution:=optional"));
+    }
+
+    @Test
+    void getVersionEndIndexTest() {
+        assertEquals(55,
+                ensureVersionMojo.getVersionEndIndex("wrap:mvn:org.apache.httpcomponents.core5/httpcore5/5.2.1"));
+        assertEquals(56, ensureVersionMojo.getVersionEndIndex(
+                "wrap:mvn:org.eclipse.californium/element-connector/3.11.0$overwrite=merge&Import-Package=net.i2p.crypto.eddsa;resolution:=optional"));
+    }
+
+    @Test
+    void getVersionTest() {
+        assertEquals("${google-oauth-client-version}", ensureVersionMojo.getVersion(
+                "wrap:mvn:com.google.oauth-client/google-oauth-client-jetty/${google-oauth-client-version}$overwrite=merge&Import-Package=com.sun.net.httpserver;resolution:=optional,*"));
```

Comment:
This should never happen since the place holders should be resolved at this point

Reply:
Yes, but I believe it's necessary to obtain the placeholder declaration even if it's not resolved, to avoid processing the bundle location.
https://github.com/apache/camel-karaf/blob/4bf0ed9588df89d3774bcb872ad61348fe75a7b5/tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java#L259
https://github.com/apache/camel-karaf/blob/4bf0ed9588df89d3774bcb872ad61348fe75a7b5/tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java#L260


then 
https://github.com/apache/camel-karaf/blob/4bf0ed9588df89d3774bcb872ad61348fe75a7b5/tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java#L159

Reply:
Maybe I could just rely on `getVersionStartIndex(Location)`

Reply:
I know that at the moment is not a good argument, but I plan to use `getVersion` in other plugin goals to be implemented in the future.